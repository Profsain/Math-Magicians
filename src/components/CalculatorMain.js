import React from 'react';
import '../styles/CalculatorMain.css';
import Footer from './Footer';
import Calculator from './Calculator';

const CalculatorMain = () => (
  <div>
    <div className="Calculator-main">
      <div className="Calculator-left">
        <h2 data-testid="headlines">Now let&apos;s do some math!</h2>
        <img src="magic-cal.jpg" alt="The magic boy doing math" />
      </div>
      <div className="Calculator-right">
        <Calculator />
      </div>
    </div>
    <Footer />
  </div>
);

export default CalculatorMain;
