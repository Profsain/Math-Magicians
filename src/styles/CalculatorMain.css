.Calculator-main {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 0 38px;
}

.Calculator-left img {
  width: 300px;
  max-width: 100%;
  margin-right: 190px;
}

.Calculator-left h2 { 
  font-size: 2rem;
  text-align: center;
}

@media only screen and (max-width: 600px) {
  .Calculator-main {
    flex-direction: column;
    padding: 0 28px;
  }

  .Calculator-right {
    width: 100%;
  }
}