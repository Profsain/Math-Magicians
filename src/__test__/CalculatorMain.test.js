import React from 'react';
import renderer from 'react-test-renderer';
import CalculatorMain from '../components/CalculatorMain';

it('renders component correctly', () => {
  const tree = renderer.create(<CalculatorMain />).toJSON();
  expect(tree).toMatchSnapshot();
});
