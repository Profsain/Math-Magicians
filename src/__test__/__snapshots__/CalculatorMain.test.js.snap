// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component correctly 1`] = `
<div>
  <div
    className="Calculator-main"
  >
    <div
      className="Calculator-left"
    >
      <h2
        data-testid="headlines"
      >
        Now let's do some math!
      </h2>
      <img
        alt="The magic boy doing math"
        src="magic-cal.jpg"
      />
    </div>
    <div
      className="Calculator-right"
    >
      <div
        className="Grid-container"
      >
        <div
          className="Result"
        >
          0
        </div>
        <button
          className="Number"
          id="ac"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="Number"
          id="plus-minus"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="Number"
          id="modul"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="Operators"
          id="div"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
        <button
          className="Number"
          id="seven"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="Number"
          id="eight"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="Number"
          id="nine"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="Operators"
          id="multi"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
        <button
          className="Number"
          id="four"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="Number"
          id="five"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="Number"
          id="six"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="Operators"
          id="minus"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <button
          className="Number"
          id="one"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="Number"
          id="two"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="Number"
          id="three"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="Operators"
          id="plus"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          className="Number Zero"
          id="zero"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="Number"
          id="dot"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="Operators"
          id="equal"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
  <footer
    className="Footer"
  >
    <p>
      Developer: Husseini Mudi @Microverse Inc
    </p>
  </footer>
</div>
`;
